src/Fortran-interface/bml_getters_m.F90

```diff
@@ -117,6 +117,7 @@ subroutine bml_get_row_single_real(a, i, row)
     ptr = bml_get_row_C(a%ptr, i - 1)
     call c_f_pointer(ptr, row_ptr, [bml_get_N(a)])
     row = row_ptr
+    deallocate(row_ptr)
```

Comment:
This fixes the valgrind test. This suggests that the assignment in https://github.com/lanl/bml/pull/348/files#diff-fc698c8987a3d4f990baad818a46a573R119 is copying the data instead of assigning a reference to it. @cnegre @jeanlucf22 @jmohdyusof @suemni what do you think?

Reply:
If it's really a data copy operation, then it's less efficient than I had hoped. Does anyone know how to get the Fortran compiler to use the reference when converting a `real(C_FLOAT), pointer` to a `real(C_FLOAT), allocatable (:)`?